packages/next/lib/page-loader.js

```diff
@@ -110,10 +123,30 @@ export default class PageLoader {
     }
   }
 
+  async prefetch (route) {
+    route = this.normalizeRoute(route)
+    const scriptRoute = route === '/' ? '/index.js' : `${route}.js`
+    if (this.prefetchCache.has(scriptRoute)) {
+      return
+    }
+    this.prefetchCache.add(scriptRoute)
+
+    const link = document.createElement('link')
+    // Feature detection is used to see if prefetch is supported, else fall back to preload
+    // Mainly this is for Safari
+    // https://caniuse.com/#feat=link-rel-prefetch
+    // https://caniuse.com/#feat=link-rel-preload
+    link.rel = supportsPrefetch ? 'prefetch' : 'preload'
+    link.href = `${this.assetPrefix}/_next/static/${encodeURIComponent(this.buildId)}/pages${scriptRoute}`
+    link.as = 'script'
```

Comment:
Consider adding the `importance` attribute to lower priority.
https://www.chromestatus.com/feature/5273474901737472

Unfortunately when this is fixed to use preload it will use the default priority for script.

Reply:
I don't think we should be using `importance` before it actually ships, as that can lock us out of that API shape if we'd want to change the API in some way. You're right that in terms of priority here, it'd use the default priority which is not ideal, but adding an `importance` attribute won't change it until it ships. I suggest we'll add it once it does.